esp-hal/src/spi/master.rs

```diff
@@ -2728,6 +2728,30 @@ pub struct Info {
 
     /// SIO3 input signal for QSPI mode.
     pub sio3_input: Option<InputSignal>,
+
+    /// SIO4 output signal for OPI mode.
+    pub sio4_output: Option<OutputSignal>,
```

Comment:
These fields shouldn't be present on devices without OPI

Reply:
Ha, `Option` isn't good enough for you :smile:. I'll see what I can do.

Reply:
Option is good enough when a single device has peripherals that support one or the other, but it's a waste of memory when the device doesn't have OPI at all. You'll likely want to macro-generate Info altogether.

Reply:
> You'll likely want to macro-generate Info altogether.

Hopefully what I pushed is what you mean.

Reply:
Almost - we should have an `opi` symbol, or add one if we don't have it.

Reply:
Ah it's "spi_octal"